### PR TITLE
Fix D values when Dmin disabled and autoadjust, coordination with new Dmin values

### PR DESCRIPTION
**Pull request by Asizon**
There is a bug since Sliders implementation that breaks D values when Dmin is disabled. up to now has used the constant values of Dmin (20-22). I have change it to works against D values again,and aplies math to calculate necessary values when Dmin changed. For the moment this D autocalculation only works with Sliders enabled and availables.

### CODE REVIEW

**Comment by Asizon** [comment]
BEFORE
dmin before

NOW
Dmindisabled

CHANGES WITHOUT DMIN
Dmindisabledchanged

CALCULATED VALUES AFTER REENABLE DMIN
Dminenabledchanged

**Comment by ctzsnooze** [comment]
This is really perfect.

Thanks, @Asizon,  for not only finding and fixing the bug, but for setting the Dmin and Dmax values correctly when the user enables Dmin from a non-default D value.

Note that the default values are Dmin 20,22 Dmax 35,35, and default normal D is 30,32.

This PR maintains those default proportions either side of the user's previous D value when they enable Dmin.

My only suggestion is that we should take this opportunity to update the default Dmin values.  They are probably a bit on the low side.  Increasing default Dmin from 20,22 to 23,25 is probably a good idea.  Slightly higher default Dmin values are likely to work better for typical freestyle builds, if anything they still are on the low side.

**Comment by mikeller** [comment]
Looks good, but we should use this to also change the default D_Min values in accordance with betaflight/betaflight#9674.

File: src/js/tabs/pid_tuning.js [hunk 1]
```diff
@@ -32,8 +32,9 @@ TABS.pid_tuning.initialize = function (callback) {
         self.activeSubtab = 'pid';
     }
 
-    // Update filtering defaults based on API version
+    // Update filtering and pid defaults based on API version
     var FILTER_DEFAULT = FC.getFilterDefaults();
+    var PID_DEFAULT = FC.getPidDefaults();
```
Comment (McGiverGim) [inline src/js/tabs/pid_tuning.js:37]: Is this being used somewhere?

Comment (mikeller) [reply src/js/tabs/pid_tuning.js:37]: Should prolly be `this.PID_DEFAULT`.

Comment (Asizon) [reply src/js/tabs/pid_tuning.js:37]: It is called in tuning-sliders.js and needed at the init to coordinate new values with sliders part. I didnt found a better way to do this

Comment (mikeller) [reply src/js/tabs/pid_tuning.js:37]: @Asizon: No. `TuningSliders.PID_DEFAULT` is called in `TuningSliders.js` - what you are doing won't work.

Comment (Asizon) [reply src/js/tabs/pid_tuning.js:37]: oh right, without this im getting sliders disabled at init :( :

image

Comment (Asizon) [reply src/js/tabs/pid_tuning.js:37]: Ok, maybe just need to call FC.getPidDefaults here?

File: src/js/fc.js [hunk 2]
```diff
@@ -692,6 +692,13 @@ var FC = {
     getPidDefaults: function() {
         var versionPidDefaults = DEFAULT_PIDS;
         // if defaults change they should go here
+        if (semver.gte(CONFIG.apiVersion, "1.43.0")) {
```
Comment (mikeller) [inline src/js/fc.js:695]: You are changing `DEFAULT_PIDS` **after** you assign it to `versionPidDefaults` - this is definitely not the way it should be.
Another no-no is that this is a getter with side-effects. Please do this in another way.

Comment (Asizon) [reply src/js/fc.js:694]: thanks, here is the problem yes, fixed now

**Comment by Asizon** [comment]
yes i understund you @mikeller  , hope that now was a good and better solution :)

**Comment by Asizon** [comment]
ok, i think this is the correct way, I was making wrong function lol:

```
    getPidDefaults: function() {
        var versionPidDefaults = DEFAULT_PIDS;
        // if defaults change they should go here
        if (semver.gte(CONFIG.apiVersion, "1.43.0")) {
            versionPidDefaults = [
                42, 85, 35, 23, 90,
                46, 90, 38, 25, 95,
                30, 90,  0,  0, 90,
            ];
        }
        return versionPidDefaults;
    },
```

**Comment by mikeller** [comment]
Oops, didn't see your last post when I hit 'Merge'. But yes, this would be an even better way - the key part is that calling a getter method like `getPidDefaults()` should not leave the system in a different state to what it was beforehand, i.e. by modifying `DEFAULT_PIDS`. Imagine if you connected to two flight controllers, the first with a firmware with API version >= 1.43, and the second with a firmware version < 1.43. What `DEFAULT_PIDS` should the second one have?

**Comment by Asizon** [comment]
okey @mikeller I can make a PR with this change, do you think it is necessary?

**Comment by mikeller** [comment]
@Asizon: It's not necessary, but it's an improvement. :-)

**Pull request merged**